### ISSUE TITLE
DeepZoom scaling quality issues

### ISSUE THREAD

**Issue by tjg1** [koush/ion#771]
I'm the lead developer of [Nori](https://github.com/tjg1/nori), an open-source image search API client for Android. Nori recently switched from okHTTP/Picasso to Ion as its HTTP and image loading library. It uses PhotoView to implement multi-touch scale and pan gestures in the full-screen image pager. 

I wanted to take advantage of DeepZoom to reduce memory footprint of the full-screen image viewer, but enabling it causes issues with scaling down high-resolution images to the PhotoView's dimensions. The lines and edges of objects look jagged, as if the image was **scaled up** from a lower resolution. This is especially noticeable with line-art and digital paintings, less so with photographs. The quality improves as the image is zoomed in.

[Example - DeepZoom enabled. PhotoView default zoom. Looks jagged.](https://my.mixtape.moe/frhyce.png)
[Example - DeepZoom enabled. PhotoView zoomed in. Looks fine.](https://my.mixtape.moe/dtgjhv.png)
[Example - DeepZoom disabled. PhotoView default zoom. Looks fine.](https://my.mixtape.moe/dlejdi.png)
[Original Image (1000x1414)](https://my.mixtape.moe/cufmsf.png)

My device's screen resolution is 1280x720. (hdpi)

You can use [Nori 3.0.0](https://github.com/tjg1/nori/releases/tag/v3.0.0) to test this, as it has DeepZoom enabled. This is the relevant image loading code: [RemoteImageFragment.java#L68](https://github.com/tjg1/nori/blob/v3.0.0/app/src/main/java/io/github/tjg1/nori/fragment/RemoteImageFragment.java#L68) 

I'm not sure if this is a duplicate of #466, as I can't view the bug report on Google Plus. [subsampling-image-view](https://github.com/davemorrissey/subsampling-scale-image-view) is another "DeepZoom" image loading library that apparently does not have this issue.

**Comment by koush** [comment]
if i recall correctly, photoview's imageview provides a hint for what tiling level deepzoom should use.

https://github.com/koush/ion/blob/master/ion-sample/src/com/koushikdutta/ion/sample/DeepZoomSample.java

See setMaximumScale

I think basically, the zoom you need to set on PhotoView has to be something like

Max.max(
image.actualWidth / PhotoView.getMeasuredWidth,
image.actualHeight / PhotoView.getMeasuredHeight,
)

Basically, you want the zoom at the maximum scale to be a 1:1 pixel mapping between the photoview and the loaded image.

**Comment by koush** [comment]
So for you, your maximum scale in that sample (1000 / 1280) is roughly 7.8125.

**Comment by koush** [comment]
What you can do is after .intoImageView, use setCallback, and then get the photoview's drawable. Grab the intrinsic with and height, and divide it by the photoview's measured width and height (as shown above) to do the max scale math.
